src/goose.rs

```diff
@@ -1222,6 +1243,28 @@ impl GooseClient {
                     }
                     raw_request.set_status_code(Some(status_code));
                     raw_request.set_final_url(r.url().as_str());
+
+                    // Load test client was redirected.
+                    if raw_request.url != raw_request.final_url {
+                        // Extract the base_url of the request made.
+                        if let Ok(original_url) = Url::parse(&raw_request.url) {
+                            if let Some(original_host) = original_url.host() {
+                                if let Some(test_base_url) = self.get_base_url().await.host() {
+                                    // Check if the base_url of the request made matches the base_url load test.
+                                    if test_base_url == original_host {
+                                        if let Ok(redirected_url) =
+                                            Url::parse(&raw_request.final_url)
+                                        {
+                                            if let Some(host) = redirected_url.host() {
```

Comment:
I feel other rust code is not using Option<> with nested ifs style wise. There feels to be a better way.

Reply:
Indeed; the way I was nesting it was completely ignoring possible errors, and therefor was wrong. Storing the base_url simplifies things, and I re-implemented this section to deal with all possible errors and without such deep nesting.